### DISCUSSION TITLE
Default form prop behaviour?

### DISCUSSION THREAD

**Discussion by ekandreas** [filamentphp/filament#1036] [Help] [answered]
When I use the ->default("123") on a filament form component the value isn't presented within the field.
I have to use the $this->form->fill() function inside "mount".
But when I do that all the data is cleared and replaced with the default values.
Shouldn't the "default" function just operate when there is no value given?

**Comment by danharrin** [comment, marked as answer]
The default value is only filled when there isn't existing data - since the defaults are usually saved to your database anyway, so this is usually a useless step.

If you'd like to fill a value into a field after it's hydrated, check out `afterStateHydrated()`:

```php
->afterStateHydrated(function ($component, $state) => {
    if (filled($state)) {
        return;
    }

    $component->state('123');
})
```
